File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Shapes/rotateshape.rb
module Asposediagramjava
  module RotateShape
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # Add a shape and set the angle
        diagram.getPages().getPage(0).getShapes().getShape(1).setAngle(190)

        # Save diagram
        diagram.save(data_dir + "RotateShape.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Rotated shape."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithShapes/RotateShape.php
<?php
namespace Aspose\Diagram\WorkingwithShapes;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class RotateShape{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram=new Diagram($dataDir."Drawing.vsd");

        # Add a shape and set the angle
        $diagram->getPages()->getPage(0)->getShapes()->getShape(1)->setAngle(190);

        # Save diagram
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."RotateShape.vdx",$saveFileFormat->VDX);
        print "Rotated shape.".PHP_EOL;
    }
}